src/measurement_buffer.cpp

```diff
@@ -152,7 +152,14 @@ void MeasurementBuffer::BufferPCLCloud(const \
 
     point_cloud_ptr cld_global(new pcl::PointCloud<pcl::PointXYZ>);
 
-    _buffer.transform(cloud, *cld_global, _global_frame);
+    // Create a tf::Transform using the tfStamped for _buffer
+    geometry_msgs::TransformStamped tf_stamped = _buffer.lookupTransform(_global_frame, cloud.header.frame_id.c_str(), ros::Time(0));
+    geometry_msgs::TransformStamped tf_stamped = _buffer.lookupTransform(_global_frame, cloud.header.frame_id.c_str(), ros::Time::now(), ros::Duration(5.0));
```

Comment:
I assume you only need one of these 2 `lookupTransform` calls? Once you resolve that, the rest of this looks good (though you need to include the header `#include <geometry_msgs/TransformStamped.h>` in `measurement_buffer.hpp`).